source/brailleTables.py

```diff
@@ -627,3 +649,52 @@ def listTables():
 # Translators: The name of a braille table displayed in the
 # braille settings dialog.
 addTable("zu-za-g2.ctb", _("Zulu grade 2"), contracted=True)
+
+# Add a new first map for the custom tables - provided in the scratchpad directory
+# and/or by addons.
+_tables = _tables.new_child()
+
+
+def initialize():
+	# The builtin tables were added at import time to the parent map.
+	# Now, add the custom tables to the first map.
+
+	import addonHandler
+	for addon in addonHandler.getRunningAddons():
+		dir_ = os.path.join(addon.path, "brailleTables")
+		if os.path.isdir(dir_):
+			tablesDirs.append(dir_)
+		try:
+			for fileName, tableConfig in addon.manifest.get("brailleTables", {}).items():
+				addTable(fileName, **tableConfig)
+		except Exception:
+			log.exception(f"Error while applying custom braille tables config from addon \"{addon.name}\"")
+
+	# Load the custom tables from the scratchpad last so it takes precedence over add-ons
+	if (
+		not globalVars.appArgs.secure
+		and config.conf['development']['enableScratchpadDir']
+	):
+		dir_ = os.path.join(config.getScratchpadDir(), "brailleTables")
+		if os.path.isdir(dir_):
+			tablesDirs.append(dir_)
+			configspec = {"brailleTables": addonHandler.AddonManifest.configspec["brailleTables"].dict()}
+			dirList = os.listdir(dir_)
+			for fileName in dirList:
+				path = os.path.join(dir_, fileName)
+				if os.path.isfile(path) and fileName.endswith(".ini"):
+					try:
+						with open(path, "rb") as file_:
+							manifest = ConfigObj(file_, configspec=configspec)
+							for fileName, tableConfig in manifest["brailleTables"].items():
+								addTable(fileName, **tableConfig)
+					except Exception:
+						log.exception(f"Error while applying custom braille tables config: {path}")
+
+
+def terminate():
+	global tablesDirs, _tables

```

Comment:
Since you just call a method on `_tables` there is no reason for it to be global.